### DISCUSSION TITLE
[Fabric] How does one create a native UI component in the new cpp-template?

### DISCUSSION THREAD

**Discussion by l3utterfly** [microsoft/react-native-windows#12958] [Q&A] [answered]
I am trying to add support for a video player in the new cpp template. Does anyone have any pointers on where/how to start?

**Comment by chrisglein** [comment, marked as answer]
Have you tried [react-native-video](https://github.com/react-native-video/react-native-video)? It supports Windows.

**Reply by l3utterfly** [reply]
Yes I have. I don't believe it supports the latest cpp-template (Fabric).

The windows code references Microsoft.Ui.Xaml.Framework element which I believe is not possible to be used in the new architecture

**Reply by l3utterfly** [reply]
I believe this is related: https://github.com/microsoft/react-native-windows/issues/12786

Except rn video IS a ui component

**Reply by chrisglein** [reply]
Ah, I missed that you were asking about Fabric. To date, we don't yet have any non-core UI components working on Fabric. It's actively under development _right now_. That's why on our [pinned Fabric issue](https://github.com/microsoft/react-native-windows/issues/12042) we don't yet invite folks to create apps targeting the new Fabric templates; getting a set of UI modules working is considered a prerequisite for considering it ready. The issue you linked (#12786) is about the non-UI modules that _should_ work today on the Fabric templates, but aren't due to some project configuration issues.

**Reply by l3utterfly** [reply]
Would you kindly point me to an implementation of a UI component template or reference such as the "Image" component or some others? I'm happy to work on a video component myself.
